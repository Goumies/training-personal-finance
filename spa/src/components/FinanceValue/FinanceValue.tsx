import React from "react";

interface Props {
  amount: number;
}

export const FinanceValue: React.StatelessComponent<Props> = ({ amount }: Props) => {
  const amountText: string = (amount >= 0 ? "+" : "") + amount.toString();
  return <span className="value">{amountText}</span>;
};
