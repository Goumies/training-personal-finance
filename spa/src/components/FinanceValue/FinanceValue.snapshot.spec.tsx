import React from "react";
import renderer from "react-test-renderer";
import { FinanceValue, Props } from "./FinanceValue";

it("should render positive amounts with a '+' sign", () => {
  const props: Props = {
    amount: 10
  };
  const tree = renderer.create(<FinanceValue {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render zero on without a sign", () => {
  const props: Props = {
    amount: 0
  };
  const tree = renderer.create(<FinanceValue {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render negative amounts with a '-' sign", () => {
  const props: Props = {
    amount: -10
  };
  const tree = renderer.create(<FinanceValue {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
