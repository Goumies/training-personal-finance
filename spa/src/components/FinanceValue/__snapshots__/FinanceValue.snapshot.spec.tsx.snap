// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render negative amounts with a '-' sign 1`] = `
<span
  className="value"
>
  -10
</span>
`;

exports[`should render positive amounts with a '+' sign 1`] = `
<span
  className="value"
>
  +10
</span>
`;

exports[`should render zero on without a sign 1`] = `
<span
  className="value"
>
  0
</span>
`;
